### PR TITLE
Fix flutter upgrade and flutter build ios

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

**Comment by Hixie** [comment]
@devoncarew 

This needed a more annoying fix because apparently we run flutter re-entrantly via xcode on iOS.

**Comment by Hixie** [comment]
(untested)

File: packages/flutter_tools/lib/src/runner/flutter_command_runner.dart [hunk 1]
```diff
@@ -147,7 +147,7 @@ class FlutterCommandRunner extends CommandRunner {
     // enginePath's initialiser uses it).
     Cache.flutterRoot = path.normalize(path.absolute(globalResults['flutter-root']));
 
-    if (globalResults['lock']);
+    if (globalResults['lock'] && !(Platform.environment['FLUTTER_ALREADY_LOCKED'] == 'true'))
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/runner/flutter_command_runner.dart]: If you pass `allowReentrantFlutter: true` in https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/commands/upgrade.dart#L54, you don't need the lock flag anymore, right?

Comment (Hixie) [inline packages/flutter_tools/lib/src/runner/flutter_command_runner.dart]: Yeah, good point.

Comment (Hixie) [inline packages/flutter_tools/lib/src/runner/flutter_command_runner.dart]: fixed

**Comment by tvolkert** [comment]
lgtm

**Pull request merged**